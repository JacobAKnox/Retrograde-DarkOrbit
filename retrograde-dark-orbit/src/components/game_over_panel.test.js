import { render } from '@testing-library/react'
import GamePanel from './game_panel';
import "setimmediate";

test('renders on-screen', () => {
  const page = render(<GamePanel/>);
  expect(page).toMatchSnapshot();
});
