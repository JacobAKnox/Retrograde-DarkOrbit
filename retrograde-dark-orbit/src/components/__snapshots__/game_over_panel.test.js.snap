// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders on-screen 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="border-4 w-full md:w-2/3 border-blue-400 min-h-[600px] bg-gray-800 md:mx-20 p-4 rounded-xl flex flex-row items-start"
      >
        <div
          class="flex flex-col w-full md:w-3/4"
        >
          <div
            class="flex flex-col w-full mb-4"
          />
          <div
            class="flex flex-grow m-2"
          >
            <div
              class="text-xl text-white text-center items-center rounded-xl m-1 flex-grow"
            >
              <div
                class="bg-slate-900 rounded-xl m-1 p-2"
              >
                Action Points:
                <br />
                0
                /
                0
              </div>
              <div
                class="grid grid-cols-3 bg-slate-900 rounded-xl m-1 p-2"
              >
                <div
                  class="rounded-xl p-2 m-2 bg-slate-700"
                >
                  <div
                    class="flex justify-center p-2 text-xl"
                  >
                    <p
                      class="text-center text-wrap break-words"
                    >
                      name
                    </p>
                  </div>
                  <div
                    class="flex justify-center p-2"
                  >
                    <div
                      class="flex rounded-xl m-1"
                    >
                      <div
                        class="flex p-1 rounded-l-xl bg-slate-900 min-w-8 justify-center hover:bg-slate-800 hover:cursor-pointer select-none text-2xl"
                      >
                        -
                      </div>
                      <div
                        class="flex bg-slate-900 p-1 min-w-12 justify-center select-none text-2xl"
                      >
                        0
                      </div>
                      <div
                        class="flex p-1 rounded-r-xl bg-slate-900 min-w-8 justify-center hover:bg-slate-800 hover:cursor-pointer select-none text-2xl"
                      >
                        +
                      </div>
                    </div>
                  </div>
                </div>
                <div
                  class="rounded-xl p-2 m-2 bg-slate-700"
                >
                  <div
                    class="flex justify-center p-2 text-xl"
                  >
                    <p
                      class="text-center text-wrap break-words"
                    >
                      name1
                    </p>
                  </div>
                  <div
                    class="flex justify-center p-2"
                  >
                    <div
                      class="flex rounded-xl m-1"
                    >
                      <div
                        class="flex p-1 rounded-l-xl bg-slate-900 min-w-8 justify-center hover:bg-slate-800 hover:cursor-pointer select-none text-2xl"
                      >
                        -
                      </div>
                      <div
                        class="flex bg-slate-900 p-1 min-w-12 justify-center select-none text-2xl"
                      >
                        0
                      </div>
                      <div
                        class="flex p-1 rounded-r-xl bg-slate-900 min-w-8 justify-center hover:bg-slate-800 hover:cursor-pointer select-none text-2xl"
                      >
                        +
                      </div>
                    </div>
                  </div>
                </div>
                <div
                  class="rounded-xl p-2 m-2 bg-slate-700"
                >
                  <div
                    class="flex justify-center p-2 text-xl"
                  >
                    <p
                      class="text-center text-wrap break-words"
                    >
                      name2
                    </p>
                  </div>
                  <div
                    class="flex justify-center p-2"
                  >
                    <div
                      class="flex rounded-xl m-1"
                    >
                      <div
                        class="flex p-1 rounded-l-xl bg-slate-900 min-w-8 justify-center hover:bg-slate-800 hover:cursor-pointer select-none text-2xl"
                      >
                        -
                      </div>
                      <div
                        class="flex bg-slate-900 p-1 min-w-12 justify-center select-none text-2xl"
                      >
                        0
                      </div>
                      <div
                        class="flex p-1 rounded-r-xl bg-slate-900 min-w-8 justify-center hover:bg-slate-800 hover:cursor-pointer select-none text-2xl"
                      >
                        +
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div
            class="w-full"
          >
            <div
              class="min-h-[200px] bg-gray-800 p-4 rounded-xl flex-col"
            >
              <div
                class="bg-gray-700 w-full h-[200px] p-2 rounded-xl overflow-y-auto"
                id="message-container"
              />
              <div
                class="bg-inherit items-center justify-start w-full h-10 mt-4 flex"
              >
                <input
                  aria-label="message"
                  class="bg-slate-700 text-white w-3/4 h-10 rounded-xl p-2 hover:bg-slate-600"
                  placeholder="message"
                  value=""
                />
                <button
                  class="bg-slate-700 text-white w-1/4 h-10 rounded-xl hover:bg-slate-600 disabled:bg-slate-950 disabled:text-gray-700"
                >
                  Send
                </button>
              </div>
            </div>
          </div>
        </div>
        <div
          class="w-full md:w-1/4 p-4 bg-gray-800 rounded-xl ml-4 pt-7"
        >
          <div
            class="bg-slate-900 text-xl text-white m-1 py-2 px-10 rounded-xl disabled:bg-slate-950 disabled:text-gray-700"
          >
            <div>
              <br />
              <p />
            </div>
          </div>
          <div
            class="bg-slate-900 text-xl text-white text-center m-1 py-2 px-10 rounded-xl"
          >
            <b
              class="text-3xl text-red-300"
            >
              <strong>
                Role
              </strong>
            </b>
            <p
              class="mb-5"
            >
              <b
                class="text-slate-200"
              >
                Name
              </b>
              <br />
            </p>
          </div>
          <style>
            
                .information-bar {
                    display: flex;
                    flex-direction: column;
                    gap: 10px;
                    background: #151B30; 
                    padding: 15px;
                    border-radius: 8px;
                    color: #fff; 
                    margin: 10px 0;
                }

                .information-bar label {
                    font-size: 0.9rem;
                    color: #FFFFFF;
                }

                .information-bar progress {
                    width: 100%;
                    height: 20px;
                    border-radius: 10px;
                    color: #4caf50;
                    background-color: #4ade80;
                }
               
                .information-bar progress::-webkit-progress-value {
                    background-color: #4caf50;
                    border-radius: 10px;
                }
     
                .information-bar progress::-webkit-progress-bar {
                    background-color: #333;
                    border-radius: 10px;
                }
                
          </style>
          <div
            class="information-bar"
          >
            <label>
              Crew: 
              <progress
                aria-label="Crew"
                max="100"
                value="50"
              />
            </label>
            <label>
              Ship Health: 
              <progress
                aria-label="Ship Health"
                max="100"
                value="50"
              />
            </label>
            <label>
              Fuel: 
              <progress
                aria-label="Fuel"
                max="100"
                value="50"
              />
            </label>
            <label>
              Life Support: 
              <progress
                aria-label="Life Support"
                max="100"
                value="50"
              />
            </label>
            <label>
              Power: 
              <progress
                aria-label="Power"
                max="100"
                value="50"
              />
            </label>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="border-4 w-full md:w-2/3 border-blue-400 min-h-[600px] bg-gray-800 md:mx-20 p-4 rounded-xl flex flex-row items-start"
    >
      <div
        class="flex flex-col w-full md:w-3/4"
      >
        <div
          class="flex flex-col w-full mb-4"
        />
        <div
          class="flex flex-grow m-2"
        >
          <div
            class="text-xl text-white text-center items-center rounded-xl m-1 flex-grow"
          >
            <div
              class="bg-slate-900 rounded-xl m-1 p-2"
            >
              Action Points:
              <br />
              0
              /
              0
            </div>
            <div
              class="grid grid-cols-3 bg-slate-900 rounded-xl m-1 p-2"
            >
              <div
                class="rounded-xl p-2 m-2 bg-slate-700"
              >
                <div
                  class="flex justify-center p-2 text-xl"
                >
                  <p
                    class="text-center text-wrap break-words"
                  >
                    name
                  </p>
                </div>
                <div
                  class="flex justify-center p-2"
                >
                  <div
                    class="flex rounded-xl m-1"
                  >
                    <div
                      class="flex p-1 rounded-l-xl bg-slate-900 min-w-8 justify-center hover:bg-slate-800 hover:cursor-pointer select-none text-2xl"
                    >
                      -
                    </div>
                    <div
                      class="flex bg-slate-900 p-1 min-w-12 justify-center select-none text-2xl"
                    >
                      0
                    </div>
                    <div
                      class="flex p-1 rounded-r-xl bg-slate-900 min-w-8 justify-center hover:bg-slate-800 hover:cursor-pointer select-none text-2xl"
                    >
                      +
                    </div>
                  </div>
                </div>
              </div>
              <div
                class="rounded-xl p-2 m-2 bg-slate-700"
              >
                <div
                  class="flex justify-center p-2 text-xl"
                >
                  <p
                    class="text-center text-wrap break-words"
                  >
                    name1
                  </p>
                </div>
                <div
                  class="flex justify-center p-2"
                >
                  <div
                    class="flex rounded-xl m-1"
                  >
                    <div
                      class="flex p-1 rounded-l-xl bg-slate-900 min-w-8 justify-center hover:bg-slate-800 hover:cursor-pointer select-none text-2xl"
                    >
                      -
                    </div>
                    <div
                      class="flex bg-slate-900 p-1 min-w-12 justify-center select-none text-2xl"
                    >
                      0
                    </div>
                    <div
                      class="flex p-1 rounded-r-xl bg-slate-900 min-w-8 justify-center hover:bg-slate-800 hover:cursor-pointer select-none text-2xl"
                    >
                      +
                    </div>
                  </div>
                </div>
              </div>
              <div
                class="rounded-xl p-2 m-2 bg-slate-700"
              >
                <div
                  class="flex justify-center p-2 text-xl"
                >
                  <p
                    class="text-center text-wrap break-words"
                  >
                    name2
                  </p>
                </div>
                <div
                  class="flex justify-center p-2"
                >
                  <div
                    class="flex rounded-xl m-1"
                  >
                    <div
                      class="flex p-1 rounded-l-xl bg-slate-900 min-w-8 justify-center hover:bg-slate-800 hover:cursor-pointer select-none text-2xl"
                    >
                      -
                    </div>
                    <div
                      class="flex bg-slate-900 p-1 min-w-12 justify-center select-none text-2xl"
                    >
                      0
                    </div>
                    <div
                      class="flex p-1 rounded-r-xl bg-slate-900 min-w-8 justify-center hover:bg-slate-800 hover:cursor-pointer select-none text-2xl"
                    >
                      +
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          class="w-full"
        >
          <div
            class="min-h-[200px] bg-gray-800 p-4 rounded-xl flex-col"
          >
            <div
              class="bg-gray-700 w-full h-[200px] p-2 rounded-xl overflow-y-auto"
              id="message-container"
            />
            <div
              class="bg-inherit items-center justify-start w-full h-10 mt-4 flex"
            >
              <input
                aria-label="message"
                class="bg-slate-700 text-white w-3/4 h-10 rounded-xl p-2 hover:bg-slate-600"
                placeholder="message"
                value=""
              />
              <button
                class="bg-slate-700 text-white w-1/4 h-10 rounded-xl hover:bg-slate-600 disabled:bg-slate-950 disabled:text-gray-700"
              >
                Send
              </button>
            </div>
          </div>
        </div>
      </div>
      <div
        class="w-full md:w-1/4 p-4 bg-gray-800 rounded-xl ml-4 pt-7"
      >
        <div
          class="bg-slate-900 text-xl text-white m-1 py-2 px-10 rounded-xl disabled:bg-slate-950 disabled:text-gray-700"
        >
          <div>
            <br />
            <p />
          </div>
        </div>
        <div
          class="bg-slate-900 text-xl text-white text-center m-1 py-2 px-10 rounded-xl"
        >
          <b
            class="text-3xl text-red-300"
          >
            <strong>
              Role
            </strong>
          </b>
          <p
            class="mb-5"
          >
            <b
              class="text-slate-200"
            >
              Name
            </b>
            <br />
          </p>
        </div>
        <style>
          
                .information-bar {
                    display: flex;
                    flex-direction: column;
                    gap: 10px;
                    background: #151B30; 
                    padding: 15px;
                    border-radius: 8px;
                    color: #fff; 
                    margin: 10px 0;
                }

                .information-bar label {
                    font-size: 0.9rem;
                    color: #FFFFFF;
                }

                .information-bar progress {
                    width: 100%;
                    height: 20px;
                    border-radius: 10px;
                    color: #4caf50;
                    background-color: #4ade80;
                }
               
                .information-bar progress::-webkit-progress-value {
                    background-color: #4caf50;
                    border-radius: 10px;
                }
     
                .information-bar progress::-webkit-progress-bar {
                    background-color: #333;
                    border-radius: 10px;
                }
                
        </style>
        <div
          class="information-bar"
        >
          <label>
            Crew: 
            <progress
              aria-label="Crew"
              max="100"
              value="50"
            />
          </label>
          <label>
            Ship Health: 
            <progress
              aria-label="Ship Health"
              max="100"
              value="50"
            />
          </label>
          <label>
            Fuel: 
            <progress
              aria-label="Fuel"
              max="100"
              value="50"
            />
          </label>
          <label>
            Life Support: 
            <progress
              aria-label="Life Support"
              max="100"
              value="50"
            />
          </label>
          <label>
            Power: 
            <progress
              aria-label="Power"
              max="100"
              value="50"
            />
          </label>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
